app/Services/Model/Imp/ScheduleService.php

```diff
@@ -0,0 +1,262 @@
+ * Copyright 2023 OpenStack Foundation
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+
+use App\Facades\ResourceServerContext;
+use App\Models\Exceptions\AuthzException;
+use App\Models\Foundation\Main\IGroup;
+use App\Models\Foundation\Summit\SelectionPlan;
+use Illuminate\Support\Facades\Log;
+use libs\utils\ITransactionService;
+use models\exceptions\EntityNotFoundException;
+use models\exceptions\ValidationException;
+use models\main\Member;
+use models\summit\ISummitEventRepository;
+use models\summit\ISummitProposedScheduleRepository;
+use models\summit\Presentation;
+use models\summit\PresentationCategory;
+use models\summit\Summit;
+use models\summit\SummitEvent;
+use models\summit\SummitProposedSchedule;
+use models\summit\SummitProposedScheduleSummitEvent;
+
+/**
+ * Class ScheduleService
+ * @package App\Services\Model
+ */
+final class ScheduleService
+extends AbstractPublishService implements IScheduleService
+{
+    /**
+     * @var ISummitProposedScheduleRepository
+     */
+    private $schedule_repository;
+
+    /**
+     * @var ISummitEventRepository
+     */
+    private $event_repository;
+
+    /**
+     * ScheduleService constructor.
+     * @param ISummitProposedScheduleRepository $schedule_repository
+     * @param ISummitEventRepository $event_repository
+     * @param ITransactionService $tx_service
+     */
+    public function __construct(ISummitProposedScheduleRepository $schedule_repository,
+                                ISummitEventRepository $event_repository,
+                                ITransactionService $tx_service)
+    {
+        parent::__construct($schedule_repository, $tx_service);
+        $this->schedule_repository = $schedule_repository;
+        $this->event_repository = $event_repository;
+    }
+
+    private function isAuthorizedUser(Member $member, Summit $summit, PresentationCategory $category): bool {
+        if ($member->isAdmin()) return true;
+        if ($member->isSummitAdmin()) return true;
+        return $summit->isTrackChair($member, $category);
+    }
+
+    /**
+     *@inheritDoc
+     */
+    public function publishProposedActivityToSource(
+        string $source, int $presentation_id, array $payload):SummitProposedScheduleSummitEvent {
+
+        $schedule = $this->tx_service->transaction(function () use ($source, $presentation_id, $payload) {
+
+            $event = $this->event_repository->getById($presentation_id);
+            if (!$event instanceof Presentation)
```

Comment:
this should be SummitEvent not presentation

Reply:
done